pallets/eth-bridge/src/tests/mock.rs

```diff
@@ -2,20 +2,40 @@ use super::*;
 use crate::{self as eth_bridge};
 use frame_support::{parameter_types, traits::GenesisBuild, BasicExternalities};
 use frame_system as system;
-use pallet_avn::{EthereumPublicKeyChecker, testing::U64To32BytesConverter};
+use pallet_avn::{testing::U64To32BytesConverter, EthereumPublicKeyChecker};
 use pallet_session as session;
 use sp_core::{ConstU32, ConstU64, H256};
 use sp_runtime::{
     testing::{Header, TestXt, UintAuthorityId},
     traits::{BlakeTwo256, ConvertInto, IdentityLookup},
     Perbill,
 };
+use sp_staking::offence::OffenceError;
 use std::cell::RefCell;
 
+thread_local! {
+    pub static OFFENCES: RefCell<Vec<(Vec<AccountId>, Offence)>> = RefCell::new(vec![]);
+}
+
 pub type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<TestRuntime>;
 pub type Block = frame_system::mocking::MockBlock<TestRuntime>;
 pub type Extrinsic = TestXt<RuntimeCall, ()>;
 pub type AccountId = u64;
+
+type IdentificationTuple = (AccountId, AccountId);
+type Offence = crate::CorroborationOffence<IdentificationTuple>;
+pub struct OffenceHandler;
+impl ReportOffence<AccountId, IdentificationTuple, Offence> for OffenceHandler {
+    fn report_offence(reporters: Vec<AccountId>, offence: Offence) -> Result<(), OffenceError> {
+        OFFENCES.with(|l| l.borrow_mut().push((reporters, offence)));
+        Ok(())
+    }
+
+    fn is_known_offence(_offenders: &[IdentificationTuple], _time_slot: &SessionIndex) -> bool {
+        false
```

Comment:
Will this change later?

Reply:
this is just basic implementation, the tests are going to come with the `eth-bridge` unit tests pr